### PR TITLE
Feature/karma

### PR DESCRIPTION
**Pull request by JeremyDezalos**
creating a new Picture in NewPicture will add karma to its attributes (set to 0)
added a function in ExistingPicture to modify the karma.

### CODE REVIEW

File: app/src/main/java/com/github/wnder/picture/Picture.java [hunk 1]
```diff
@@ -102,4 +103,20 @@ public void onUpdatedGuessesAvailable(Consumer<Map<String, Location>> guessesAva
...
+     * Apply consumer function when the karma is available
+     * @param karmaAvailable consumer function to call when the karma is available
+     */
+    public void onKarmaUpdated(Consumer<Map<String, Object>> karmaAvailable){
+        Task<DocumentSnapshot> karmaTask = Storage.downloadFromFirestore("pictures", uniqueId);
+        karmaTask.addOnSuccessListener((documentSnapshot) -> {
+            Map<String, Object> map = new HashMap<>();
+            //We put these attributes back because if we don't, they disappear from the db
+            map.put("latitude", documentSnapshot.getDouble("latitude"));
+            map.put("longitude", documentSnapshot.getDouble("longitude"));
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/picture/Picture.java:117]: There might be a way to do this with firestore (add/update fields without deleting the others).

**Review by rkuenzi-epfl — APPROVED** [review]
It think there should be some way to update fields on firestore but this would need to change the Storage class. Otherwise it's fine as it is works now.

**Comment by JeremyDezalos** [comment]
> It think there should be some way to update fields on firestore but this would need to change the Storage class. Otherwise it's fine as it is works now.

Agreed, an update function would be nice

**Comment by JeremyDezalos** [comment]
Since Romain has started his branch from this branch, there is no need to merge this

**Review by BoostedBoat — APPROVED** [review]
Warning: Firestore has to be regularly cleaned to delete pictures without karma to avoid future problems, otherwise everything seems good to me.

**Pull request merged**